openfe/protocols/openmm_septop/equil_septop_method.py

```diff
@@ -0,0 +1,2125 @@
+          timeseries for each repeat of that simulation type.
+        """
+        replica_states: dict[str, list[npt.NDArray]] = {
+            'complex': [], 'solvent': []
+        }
+
+        def is_file(filename: str):
+            p = pathlib.Path(filename)
+
+            if not p.exists():
+                errmsg = f"File could not be found {p}"
+                raise ValueError(errmsg)
+
+            return p
+
+        def get_replica_state(nc, chk):
+            nc = is_file(nc)
+            dir_path = nc.parents[0]
+            chk = is_file(dir_path / chk).name
+
+            reporter = multistate.MultiStateReporter(
+                storage=nc, checkpoint_storage=chk, open_mode='r'
+            )
+
+            retval = np.asarray(reporter.read_replica_thermodynamic_states())
+            reporter.close()
+
+            return retval
+
+        for key in ['complex', 'solvent']:
+            for pus in self.data[key].values():
+                states = get_replica_state(
+                    pus[0].outputs['nc'],
+                    pus[0].outputs['last_checkpoint'],
+                )
+                replica_states[key].append(states)
+
+        return replica_states
+
+    def equilibration_iterations(self) -> dict[str, list[float]]:
+        """
+        Get the number of equilibration iterations for each simulation.
+
+        Returns
+        -------
+        equilibration_lengths : dict[str, list[float]]
+          Dictionary keyed `complex` and `solvent` for each leg
+          of the thermodynamic cycle, with lists containing the
+          number of equilibration iterations for each repeat
+          of that simulation type.
+        """
+        equilibration_lengths: dict[str, list[float]] = {}
+
+        for key in ['complex', 'solvent']:
+            equilibration_lengths[key] = [
+                pus[0].outputs['equilibration_iterations']
+                for pus in self.data[key].values()
+            ]
+
+        return equilibration_lengths
+
+    def production_iterations(self) -> dict[str, list[float]]:
+        """
+        Get the number of production iterations for each simulation.
+        Returns the number of uncorrelated production samples for each
+        repeat of the calculation.
+
+        Returns
+        -------
+        production_lengths : dict[str, list[float]]
+          Dictionary keyed `complex` and `solvent` for each leg of the
+          thermodynamic cycle, with lists with the number
+          of production iterations for each repeat of that simulation
+          type.
+        """
+        production_lengths: dict[str, list[float]] = {}
+
+        for key in ['complex', 'solvent']:
+            production_lengths[key] = [
+                pus[0].outputs['production_iterations']
+                for pus in self.data[key].values()
+            ]
+
+        return production_lengths
+
+    def restraint_geometries(self) -> tuple[list[dict[str, Any]], list[dict[str, Any]]]:
```

Comment:
Could we return the `BoreschRestraintGeometry` objects they are probably easier to work with and should contain all the information we need?

Reply:
I _think_ we discussed this in the past, the issue at the moment is that RestraintGeometry isn't a GufeTokenizable. It probably should be, but it might be better for us to deal with this as a separate issue.

Reply:
Is it worth just adding to the restraint API PR?

Reply:
Separate issue please, turning it into a gufetokenizable is non negligible amounts of work and I'd prefer we deal with it after merge.